src/main/kotlin/org/openrewrite/kotlin/KotlinTypeMapping.kt

```diff
@@ -878,11 +879,13 @@ class KotlinTypeMapping(typeCache: JavaTypeCache, firSession: FirSession) : Java
         val annotations = listAnnotations(symbol.annotations)
         var resolvedOwner: JavaType? = owner
         if (owner == null) {
-            val lookupTag: ConeClassLikeLookupTag? = symbol.containingClassLookupTag()
-            if (lookupTag != null && lookupTag.toSymbol(firSession) !is FirAnonymousObjectSymbol) {
-                // TODO check type attribution for `FirAnonymousObjectSymbol` case
-                resolvedOwner =
-                    type(lookupTag.toFirRegularClassSymbol(firSession)!!.fir) as JavaType.FullyQualified?
+            // TODO: fix type attribution for anonymous objects. Currently, returns JavaType.Unknown.
+            if (symbol.dispatchReceiverType != null) {
+                resolvedOwner = type(symbol.dispatchReceiverType)
```

Comment:
Just for context: What I did was to move these lines from the parser here, so that we don't need to duplicate this logic for the PSI-based parser: https://github.com/openrewrite/rewrite-kotlin/commit/6e46b0ea00990f1c8c90f05df951aefa399eb1cf#diff-0f192ceb739be61c09f94d0991022e59b90560191b2760a9b297692ccaac212dL4748-L4754

Reply:
Ahhh, that makes sense -- tysm!